src/dotty/tools/dotc/transform/VCInlineEqAndField.scala

```diff
@@ -7,20 +7,18 @@ import Contexts._, Trees._, StdNames._, Symbols._
 import DenotTransformers._, TreeTransforms._, Phases.Phase
 import ExtensionMethods._, TreeExtractors._, ValueClasses._
 
-/** This phase inlines calls to methods and fields of value classes.
+/** This phase inlines calls to == and to the field of a value class
  *
  *  For a value class V defined as:
- *    case class V(val underlying: U) extends AnyVal
- *  We replace method calls by calls to the corresponding extension method:
- *    v.foo(args) => V.foo$extension(v.underlying(), args)
- *  And we avoid unnecessary allocations:
+ *    class V(val underlying: U) extends AnyVal
+ *  we avoid unnecessary allocations:
  *     new V(u1) == new V(u2) => u1 == u2
  *    (new V(u)).underlying() => u
  */
-class VCInline extends MiniPhaseTransform with IdentityDenotTransformer {
+class VCInlineEqAndField extends MiniPhaseTransform with IdentityDenotTransformer {
   import tpd._
 
-  override def phaseName: String = "vcInline"
+  override def phaseName: String = "vcInlineEqAndField"
```

Comment:
>  == is in no way special, and should be treated as other methods. It does not need any special treatment at all.

`==` handling before this PR is special and the name of the phase reflects that, we may change how we deal with `==` but that will be in a separate PR. Note that the special handling of `==` is only for the `==` defined in `Any`, if you define your own `==` then it will behave like a normal method.

> So what this phase does is peephole-optimization of box-unbox pattern. It is not field inlining(vcInlineField), which is a separate notion defined in papers and has nothing to do with value classes.

OK, maybe field inlining has another meaning, but I'd still like it if the name of the phase described what the phase does, what about `VCInlineGetter` or `VCElideBoxUnbox`?